src/components/SRNEntry.vue

```diff
@@ -78,49 +89,73 @@ export default {
       return this.redirectTo == "plio";
     },
     isSubmitButtonDisabled() {
-      return !this.isAnyUserIDPresent || this.invalidInputMessage != "";
+      //submit button is disabled if no SRN has been typed or input is invalid or SRN hasn't been completely typed
+      return (
+        !this.isAnyUserIDPresent ||
+        this.invalidInputMessage != "" ||
+        this.userIDList.slice(-1)[0]["userID"].length < this.maxLengthOfSRN - 1
+      );
     },
     isUserValidated() {
       //if multiple SRN's are allowed, then this helps with the activation of the + button, to add more SRN's
       return (
         !this.isSingleEntryOnly &&
-        this.userIDList[0]["userID"].length > this.maxLengthOfSRN - 1
+        this.userIDList.slice(-1)[0]["userID"].length > this.maxLengthOfSRN - 1 &&
+        this.userIDList.length < 10
       );
     },
   },
   methods: {
     isValidSRNFormat(e) {
       //checking to see if each char typed by user is only a number
-      let char = String.fromCharCode(e.keyCode);
-      if (/^[0-9]+$/.test(char)) return true;
+      if (e.keyCode >= 48 && e.keyCode <= 57) return true;
       else e.preventDefault();
     },
-    addField(value, list) {
-      list.push({ value: "" });
+    async addField(list) {
+      //for adding another field, the previously entered ID is validated against the database
+      const userID = parseInt(this.userIDList.slice(-1)[0]["userID"]);
```

Comment:
- How about making this a computed variable? `latestEntry`? That way you can just use it like
`const userID = parseInt(this.latestEntry)`

- also, 
```suggestion
      const latestUserID = parseInt(this.userIDList.slice(-1)[0]["userID"]);
```